frontend/apps/crates/entry/jig/edit/src/edit/sidebar/module/actions.rs

```diff
@@ -144,7 +148,17 @@ pub fn assign_kind(state: Rc<State>, kind: ModuleKind) {
                     id,
                     kind,
                 }));
-                state.sidebar.modules.lock_mut().set_cloned(index, module);
+
+                {
+                    // Instead of replacing the module at the index, we remove the old module and
+                    // add the new one. This is slightly less efficient because it fires signals
+                    // for the entire list of modules, however, it is necessary so that the modules
+                    // before and after this one can have their views updated.
+                    let mut modules = state.sidebar.modules.lock_mut();
+                    modules.remove(index);
+                    modules.insert_cloned(index, module);
+                }
```

Comment:
this change was necessary to hide the add button for modules when the next module in the list is a placeholder. 

If it's not necessary to hide the add button in those cases, this change can be reverted and probably [this code](https://github.com/ji-devs/ji-cloud/pull/1967/files#diff-78f7288368221bbdbdf08c195a5678f38c88c7f3dfa25723e664c25ccdd2cf2dR49-R66) can be simplified or removed.

Reply:
I think that this change is good even if it adds a bit of complexity